src/components/pages/Publish/index.tsx

```diff
@@ -133,14 +133,16 @@ export default function PublishPage({
         'Publish with ',
         metadata,
         serviceType,
-        values.dataTokenOptions
+        values.dataTokenOptions,
+        values.provider === 'Custom' ? values.customProvider : null
       )
 
       const ddo = await publish(
         metadata as unknown as Metadata,
         serviceType,
         values.dataTokenOptions,
-        timeout
+        timeout,
+        values.provider === 'Custom' ? values.customProvider : null
```

Comment:
right, now I get the capturing of 2 values to differentiate between default & custom selection. Maybe it would reduce some logic in some places if we only rely on an optional `customProvider` key which we could switch between a string and `null` depending on user selection which would reduce need to have 2 different keys. But in end also no big deal as we do not send this data around in end anyway

Reply:
I think you're right, it is better with just one field `customProvider`, I don;t think there is any need for the provider box selector when this is hidden behind advanced settings. I have removed the provider field